src/libcore/option.rs

```diff
@@ -1107,3 +1107,29 @@ impl<A, V: FromIterator<A>> FromIterator<Option<A>> for Option<V> {
         }
     }
 }
+
+/// The error type that results from applying the try operator (`?`) to a `None` value. If you wish
+/// to allow `x?` (where `x` is an `Option<T>`) to be converted into your error type, you can
+/// implement `impl From<NoneError>` for `YourErrorType`. In that case, `x?` within a function that
+/// returns `Result<_, YourErrorType>` will translate a `None` value into an `Err` result.
+#[unstable(feature = "try_trait", issue = "42327")]
+#[derive(Clone, Copy, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
+pub struct NoneError;
```

Comment:
Should we implement `std::error::Error` for this type?

Reply:
Will this be exposed insta-stably as `<Option<()> as Try>::Error`?  Would that imply it should be `#[stable]`?

Reply:
The *existence* of a such a type is basically exposed, but not the precise name of it, I think. So I think it's ok to mark it unstable still.

Reply:
Oh, and, I think the `Try` trait is still marked as unstable, so actually that name is not stable either.